### ISSUE TITLE
Linking fails on MacOSX (missing -luuid)

### ISSUE THREAD

**Issue by hrydgard** [Traverse-Research/hassle-rs#23]
I'm trying (in a non-embark-related side project) to use this from a build.rs on a Macbook Air M1, but any use of hassle-rs causes a linker error about a missing -luuid when trying to build the build.rs:

>
> error: linking with `cc` failed: exit status: 1
>  |
  = note: "cc" "-arch" "arm64" "/Users/hrydg/dev/rgb/target/debug/build/rgb-dca230a1aa5b7e54/build_script_build-dca230a1aa5b7e54.10m9mxptggm1u1hd.rcgu.o" "/Users/hrydg/dev/rgb/target/debug/build/rgb-dca230a1aa5b7e54/build_script_build-dca230a1aa5b7e54.14wrnmxnrm9y2a17.rcgu.o" "/Users/hr
>
> [SNIP]
>
>libcompiler_builtins-56bb43a121401f7f.rlib" "-luuid" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/hrydg/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/Users/hrydg/dev/rgb/target/debug/build/rgb-dca230a1aa5b7e54/build_script_build-dca230a1aa5b7e54" "-Wl,-dead_strip" "-nodefaultlibs" "-undefined" "dynamic_lookup"
  = note: ld: library not found for -luuid
          clang: error: linker command failed with exit code 1 (use -v to see invocation)

Basically, seems like some library that hassle depends on is trying to link with -luuid, which apparently is not a thing on Mac (libuuid is header-only, apparently).

(note: just thought it might be good to report, you probably should not give this any kind of priority, heh)

**Comment by MarijnS95** [comment]
This is the ancient `com` library.

There is a replacement, Microsoft's `com-rs` crate, and the conversion sits on the `ms-com-rs` branch here - that should resolve this linker error (it solves the exact same one on Android).

Unfortunately, just as I was planning to clean up and PR the conversion, Microsoft announced its `windows-rs` crate with COM interop, auto-generated bindings and all. To this day there is still no consensus on whether `com-rs` will live on and be maintained (it's still sees occasional fixes and releases), or whether it will be superseded by `windows-rs` which carries lots of similar code and intent. The latter situation requires a few minor changes to be carried by `windows-rs` to support Linux (and Mac) even if we just need some generated bindings and the COM vtable boilerplate.

**Comment by hrydgard** [comment]
Ok, well, maybe let's see how that shakes out, and I can play with the `ms-com-rs` branch in the meantime, thanks for letting me know about it!

**Comment by expenses** [comment]
Installing `ossp-uuid` (`brew install ossp-uuid`) fixed this for me.

**Comment by hrydgard** [comment]
@expenses  are you on an x86 mac? On M1, installing that did not fix it for me.

**Comment by expenses** [comment]
I am, yeah.
